pkgs/development/libraries/science/math/or-tools/default.nix

```diff
@@ -1,30 +1,71 @@
-{ lib, stdenv, fetchFromGitHub, cmake, abseil-cpp, gflags, which
-, lsb-release, glog, protobuf, cbc, zlib
-, ensureNewerSourcesForZipFilesHook, python, swig }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, abseil-cpp
+, bzip2
+, zlib
+, lsb-release
+, which
+, protobuf
+, cbc
+, ensureNewerSourcesForZipFilesHook
+, python
+, swig4
+}:
 
 stdenv.mkDerivation rec {
   pname = "or-tools";
-  version = "7.7";
+  version = "8.1";
+  disabled = python.pythonOlder "3.6";  # not supported upstream
 
   src = fetchFromGitHub {
     owner = "google";
     repo = "or-tools";
     rev = "v${version}";
-    sha256 = "06ig9a1afmzgzcg817y0rdq49ahll0q9y7bhhg9d89x6zy959ypv";
+    sha256 = "1zqgvkaw5vf2d8pwsa34g9jysbpiwplzxc8jyy8kdbzmj8ax3gpg";
   };
 
+  patches = [
+    # This patch (on master as of Feb 11, 2021) fixes or-tools failing to respect
+    # USE_SCIP=OFF and then failing to find scip/scip.h
+    (fetchpatch {
+      url = "https://github.com/google/or-tools/commit/17321869832b5adaccd9864e7e5576122730a5d5.patch";
+      sha256 = "0bi2z1hqlpdm1if3xa5dzc2zv0qlm5xi2x979brx10f8k779ghn0";
+    })
+  ];
+
   # The original build system uses cmake which does things like pull
   # in dependencies through git and Makefile creation time. We
   # obviously don't want to do this so instead we provide the
   # dependencies straight from nixpkgs and use the make build method.
   configurePhase = ''
     cat <<EOF > Makefile.local
-    UNIX_ABSL_DIR=${abseil-cpp}
-    UNIX_GFLAGS_DIR=${gflags}
-    UNIX_GLOG_DIR=${glog}
-    UNIX_PROTOBUF_DIR=${protobuf}
-    UNIX_CBC_DIR=${cbc}
+      UNIX_ABSL_DIR=${abseil-cpp}
+      UNIX_PROTOBUF_DIR=${protobuf}
+      UNIX_CBC_DIR=${cbc}
+      USE_SCIP=OFF
     EOF
+
+    # bzip2 is required in one of the tests but not on the link line
+    substituteInPlace makefiles/Makefile.unix.mk \
+      --replace 'ZLIB_LNK = -lz' 'ZLIB_LNK = -lz -lbz2'
```

Comment:
Is this perhaps a function of our `cbc` being linked with bzip2, while the upstream bundled one that we don’t use is configured with `--disable-bzlib`?

Reply:
Yeah.

```console
$ grep -r bz2 /nix/store/72lphip7y1vfhj2fnb0328ghs585qfrq-cbc-2.10.4/lib/pkgconfig
/nix/store/72lphip7y1vfhj2fnb0328ghs585qfrq-cbc-2.10.4/lib/pkgconfig/coinutils.pc:Libs: -L${libdir} -lCoinUtils -lbz2 -lz -lm 
```

I think this means the `-lbz2` belongs in `COINUTILS_LNK`? (Upstream really should be using `pkg-config` instead…)

Reply:
This makes sense, but I think I don't understand make syntax well enough to make it work?

I tried
```
substituteInPlace makefiles/Makefile.third_party.unix.mk \
      --replace 'COINUTILS_LNK = $(STATIC_COINUTILS_LNK)' \
                'COINUTILS_LNK = "$(STATIC_COINUTILS_LNK) -lbz2"'
```
and it all just blows up...

Reply:
Fixed